patches/minecraft/net/minecraft/client/Minecraft.java.patch

```diff
@@ -352,20 +380,21 @@
 -            }
 -
 -         }
++         if (!net.minecraftforge.client.ForgeHooksClient.onClickInput(2, this.field_71474_y.field_74322_I, null).isCanceled())
```

Comment:
You're better off just putting MAIN_HAND instead of null, otherwise you should put @Nullable on the getHand method. The problem with that though is every modder will have to create null check when it could simply be avoid completely by putting MAIN_HAND there instead.

Reply:
I wanted to make it ``@Nullable`` but I couldn't determine which dependency's version Forge uses. (as there are multiple dependencies with their own ``@Nullable``) It doesn't really matter if it's null or MAIN_HAND though so I've changed it to MAIN_HAND now, probably easier than giving people NullPointerExceptions.

Reply:
Null pointers was the curse of Minecraft before they added ItemStack.EMPTY. It actually a really good practice.